subsys/usb/usb_device.c

```diff
@@ -971,6 +986,10 @@ static void forward_status_cb(enum usb_dc_status_code status, const u8_t *param)
 		}
 	}
 
+	if (usb_dev.user_status_callback) {
```

Comment:
Was this intended to map `enum usb_dc_status_code` to `enum usb_state`?  If not, we can drop the commit that adds that enum.

Reply:
[usb_dc_status_code](https://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/usb/usb_dc.h#L41) should be reordered

Reply:
Adding this enum I intended to provide to user the **state** enum type which will match states described in spec and nothing more. The callback is still using **statuses** as my intention was that the user is responsible for controlling the state.

Reply:
> [usb_dc_status_code](https://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/usb/usb_dc.h#L41) should be reordered

Do You mean the order of statuses should match those from [usb_state](https://github.com/zephyrproject-rtos/zephyr/pull/20375/files#diff-0a5e6a18914d2e2bf1bf9595e8ab483eR105) ?

Reply:
Yes, otherwise it will not work

Reply:
Right now I dont think there is a need to change anything in the  [usb_dc_status_code](https://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/usb/usb_dc.h#L41) as I agreed to remove usb_device_state (justification below)

Reply:
:confused: But was it not what @carlescufi and you wanted?

Reply:
What we wanted is to expose the Device States from the spec to the application. I disagree with @pabigot here, and I think we should expose them all. I see no need to just expose a subset.

Reply:
To clarify: my objection was to adding `enum usb_state` as a distinct type but not providing the translation from the existing DC states to them when passing them to the callback.  If you want to do the translation, that's fine, but from what I understand it would only work on Nordic because the other implementations can't provide the necessary detail.

Also, I do believe that "disconnected" is a state I need to be informed about (distinct from suspended), even if the spec doesn't consider it a named "device state".